graphql/util.go

```diff
@@ -1290,3 +1290,45 @@ func getAPISubscriptionsForProject(ctx context.Context, projectId string) ([]*re
 	}
 	return res, nil
 }
+
+func GroupProjects(projects []model.ProjectRef, onlyDefaultedToRepo bool) ([]*GroupedProjects, error) {
+	groupsMap := make(map[string][]*restModel.APIProjectRef)
+
+	for _, p := range projects {
+		var groupName string
+		if onlyDefaultedToRepo {
```

Comment:
disregard, I understand what's happening now. I think setting groupName here and then setting it again below is a bit confusing. Can I suggest setting groupName to `owner/repo` here unless `onlyDefaultedToRepo && !p.UseRepoSettings` in which case we set to the empty string? Then we can skip the re-processing later on

Reply:
that won't work because the sorting needs to happen based on the repoRefId for the onlyDefaultedToRepo. It sorts it into the based on the repoRefId and then later on changes it to the human readable form.

Reply:
In other words, I can't skip the re-proccessing because we need it to be the repoRefId at first and the displayname (owner/repo) later.

Reply:
But repoRefId is created based on owner/repo; it's essentially the same thing.

Reply:
That is, when we assign a repo ref to a project ref, we look for a corresponding owner/repo pair, and if it exists we assign the RepoRefId and if it doesn't we make a new one. 
https://github.com/evergreen-ci/evergreen/blob/f2137ddff03f27ca606dffe523b0d5a3d762c25c/model/project_ref.go#L455

Reply:
ahh good point, so I can use owner/repo as the key instead. I'll update it. thanks!